java/Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM eclipse-temurin:17.0.6_10-jdk
+
+WORKDIR /lib
+
+# Install protoc (cf. http://google.github.io/proto-lens/installing-protoc.html)
+ENV PROTOC_ZIP=protoc-21.12-linux-x86_64.zip
+RUN apt-get update && apt-get install -y unzip
+RUN curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v21.12/$PROTOC_ZIP \
+    && unzip -o $PROTOC_ZIP -d /usr/local bin/protoc \
+    && unzip -o $PROTOC_ZIP -d /usr/local 'include/*' \
+    && rm -f $PROTOC_ZIP
```

Comment:
I'm wondering if we can install `protoc` via `apt-get` as well?  Aka https://packages.debian.org/bullseye/protobuf-compiler

Might be more concise

Reply:
Trying to install with `apt-get` seems to introduce a bunch of additional dependencies that (IMO) would just be bloat here. Given that, I'd prefer to keep things simple and continue using the existing installation pattern

Reply:
@bdferris-v2 I'm running into a couple issues here.
- it looks like the default `maven` docker image uses Ubuntu. On [Ubuntu's package repo](https://packages.ubuntu.com/jammy/protobuf-compiler), I'm only able to find a significantly older version of `protobuf-compiler` (3.12.4). It looks like this also causes a diff in the codegen, so I would prefer to use a more up-to-date version of the compiler. Not sure if someone can push up a more recent version of that to Ubuntu?
- I tried to switch over to `maven`'s `alpine` image, but I'm running into a number of issues like `apk` not being able to find the `protobuf-compiler` package for some reason `apk add protobuf-compiler`. I'm also fighting with getting Alpine docker images to play nicely with M1 Apple Silicon hardware :(

Reply:
From what I can tell, the least platform dependent way (and most fine-grained as far as maintaining a consistent protoc version) is to install still seems to be installing from source rather than using a package manager. What do you think about keeping it that way?

Reply:
I guess the old protobuf-compiler images on Unbuntu jammy are probably reason enough to avoid that path and install the binary directly.  LGTM then.